MVP guidelines for Numerical Input (#2448)
coveo/plasma: packages/website/docs/NumericInput.md

# Best Practices

Use a numeric input to allow users to make **small and precise incremental changes**.

A numeric input is especially appropriate when users must select a value within a short range. If your range has no limit, or if the limit is arbitrary, consider removing the incremental stepper.

A numeric input should be wide enough to display the maximum number of digits a value can have.

# Labeling

Keep labels short, preferably **under three words**.

Provide a descriptive title without action verbs. For example, write "Price" rather than "Enter a price".

# Help Text and Instructions

If the value to select must be within a range, consider indicating the minimum and maximum values, or only one of them, in a help text (e.g., "Maximum of 20").

# Feedback and Validation

Ideally, the value to enter in a numerical input should be an integer. If you expect users to frequently enter decimals, **show the decimal point** in the default value (e.g.:, "20.0") to indicate the accepted format.

Display a validation message when users enter a value outside of the expected range. This message should include the maximal and minimal values accepted.

> Determine the stepper increment relative to the range. An increment of 10% is generally acceptable. For instance, if the range is 1-10, the increments should be of 1. If the range is 1-5000, the increments of 500 would be acceptable.

Reviewer: ```suggestion
Determine the stepper increment relative to the range. An increment of 10% is generally acceptable. For instance, if the range is 1-10, the increments should be of 1. If the range is 1-5000, increments of 500 would be acceptable.
```
@kpoulincoveo This is my last suggestion!

# Best Practices

Use a numeric input when the user needs **small and precise** incremental changes.

A numeric input is more appropriate when the value is set within a shorter range. If your value has no limit, or if the limit is arbitrary, consider removing the incremental stepper.

The width of a numeric input should be large enough to display the maximum number of digits.

## Labeling

Keep labels short, preferably **under three words.**

Provide a descriptive title without action verbs. For example, write "Price" rather than "Enter your price".

## Help Text and Instructions

If the input is set within a range, consider indicating the minimum and maximum values, or only one of them, in a help text, e.g. "Maximum of 20".

## Feedback and Validation

Numerical inputs are best when the value doesn't require decimal or fraction. If decimals are expected to be frequently set by users, make sure to show **the decimal point** in the default value, e.g.: "20.0", to indicate which format is accepted.

> Always shows in the validation message both the minimal and maximal limit if the user tries to enter a number outside the acceptable range.

Reviewer: ```suggestion
Display a validation message when users enter a value outside of the expected range. This message should include the maximal and minimal values accepted.
```

> Determine the stepper increment relative to the range. For instance, if the range is 1-10, the increments should be of 1. If the range is 1-100, the increments should be of 10.

Reviewer: ```suggestion
Determine the stepper increment relative to the range. For instance, if the range is 1-10, the increments should be of 1. If the range is 1-100, the increments should be of 10.
```
@kpoulincoveo Not sure I understand the advice here. Can we be more specific? Should we recommend increments of 10%?

Author: I think we can suggest an increment of 10% but am not sure it always gonna be the case.

---

## Related Components

> If your use case doesn't match the guidelines above, consider using one of the following components instead:

Reviewer: ```suggestion
If value precision is unimportant, consider using a [slider](#/form/Sliders) instead.
```

Reviewer 2: @kpoulincoveo

> -   [Sliders](#/form/Sliders) when the precision of the value is not important.

Reviewer: Delete if you accept the above suggestion

Reviewer 2: @kpoulincoveo

Author: It was updated
